### PR TITLE
feat: add custom header param support

### PR DESCRIPTION
**Pull request by eshanholtz**
# Fixes #

This PR adds the open api specifications for custom header parameters supported by Twilio API endpoints, as well as additional relevant debugger output. 

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the master branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: src/services/cli-http-client.js [hunk 1]
```diff
@@ -125,6 +125,17 @@ class CliRequestClient {
       this.logger.debug(options.params);
     }
 
+    // Deep copy headers since we'll be modifying
+    let customHeaders = JSON.parse(JSON.stringify(options.headers));
+    const standardHeaders = ['User-Agent', 'Accept-Charset', 'Connection', 'Authorization', 'Accept', 'Content-Type'];
```
Comment (childish-sambino) [inline src/services/cli-http-client.js:130]: nit: Should define this as const at top of file. See `NETWORK_ERROR_CODES`.

File: src/services/cli-http-client.js [hunk 2]
```diff
@@ -125,6 +125,17 @@ class CliRequestClient {
...
+    standardHeaders.forEach(header => {
+      delete customHeaders[header];
```
Comment (childish-sambino) [inline src/services/cli-http-client.js:132]: I'm paranoid about something else using this client and using a lower-cased `authorization` header (unlikely, but possible) and it getting logged. Prefer to do case-agnostic filtering.

Comment (eshanholtz) [reply src/services/cli-http-client.js]: I like this new solution.

File: src/services/cli-http-client.js [hunk 3]
```diff
@@ -8,6 +8,8 @@ const { NETWORK_ERROR } = require('../services/messaging/help-messages');
 
 const NETWORK_ERROR_CODES = new Set(['ETIMEDOUT', 'ESOCKETTIMEDOUT', 'ECONNABORTED']);
 
+const STANDARD_HEADERS = ['user-agent', 'accept-charset', 'connection', 'authorization', 'accept', 'content-type'];
```
Comment (childish-sambino) [inline src/services/cli-http-client.js:11]: nit: use a set for efficiency (probably)

**Pull request merged**